### PR TITLE
Connection: Check if user is connected before attempting to initialize analytics with a user in useProductCheckoutWorkflow hook

### PR DESCRIPTION
**Pull request by oskosk**
Fixes bug introduced in https://github.com/Automattic/jetpack/pull/26964

image

#### Changes proposed in this Pull Request:

* Checks for connection before initializing JS analytics library

#### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?

#### Does this pull request change what data or activity we track or use?

Yes and no. It fixes something in one analytics hook.

#### Testing instructions:

* Go to '..'
*

### CODE REVIEW

File: projects/js-packages/connection/hooks/use-product-checkout-workflow/index.jsx [hunk 1]
```diff
@@ -41,7 +41,7 @@ export default function useProductCheckoutWorkflow( {
 	} );
 
 	const initializeAnalytics = () => {
-		const tracksUser = syncSelect( STORE_ID ).getWpcomUser();
+		const tracksUser = isUserConnected ? syncSelect( STORE_ID ).getWpcomUser() : null;
 		const blogId = syncSelect( STORE_ID ).getBlogId();
```
Comment (retrofox) [inline projects/js-packages/connection/hooks/use-product-checkout-workflow/index.jsx]: why do you continue running the following instructions? You should exit from here asap.

Comment (oskosk) [reply projects/js-packages/connection/hooks/use-product-checkout-workflow/index.jsx]: Just force-pushed again with a different approach

**Review by retrofox — APPROVED** [review]
LGTM

**Comment by nateweller** [comment]
Thanks for jumping on this fix so quick @oskosk 🥇 

I am still running into the `.getWpcomUser is not a function` error after pulling trunk into https://github.com/Automattic/jetpack/pull/26069. 

Screen Shot 2022-10-20 at 4 55 45 PM

When I log out `isUserConnected`, it is `true`. However, `syncSelect( STORE_ID )` only returns the following methods:

Screen Shot 2022-10-20 at 4 57 27 PM

I'm a bit confused as I only see `getWpcomUser` selectors in the `search` and `wordads` packages, do you know where this selector is being defined? 

cc @robfelty

**Comment by oskosk** [comment]
Thanks @nateweller I'm checking now

**Comment by retrofox** [comment]
> I'm a bit confused as I only see getWpcomUser selectors in the search and wordads packages, do you know where this selector is being defined?

That's a good point. The selector is not defined in the jetpack-connection store.

**Comment by nateweller** [comment]
Looking at the next line as well, I think the `getBlogId()` will also likely have the same problem.

**Comment by jsnmoon** [comment]
What are your thoughts about reverting #26964's changes to the connection package? I don't think it's worth introducing render-breaking bugs to add analytics to a generic function.

**Comment by retrofox** [comment]
https://github.com/Automattic/jetpack/pull/26978 is RfR cc @jsnmoon @nateweller

**Comment by oskosk** [comment]
>What are your thoughts about reverting https://github.com/Automattic/jetpack/pull/26964's changes to the connection package? I don't think it's worth introducing render-breaking bugs to add analytics to a generic function.

I would normally do that @jsnmoon. Indeed

**Pull request merged**